File: 169-majority-element/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        map<int,int>mpp;
        for(auto & it:nums){
            mpp[it]++;
        }
        for(auto & it :mpp){
            if(it.second>nums.size()/2){
                return it.first;
            }
        }
        return -1;
    }
};

File: 169-majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        Map<Integer,Integer>mpp= new HashMap<>();
        for(int  it :nums){
            if(mpp.containsKey(it)){
                mpp.put(it,mpp.get(it)+1);
            }
            else{
                mpp.put(it,1);
            }
        }
        for(int key : mpp.keySet()){
            int val=mpp.get(key);
            if(val>nums.length/2){
                return key;
            }
        }
        return -1;
    }
}